### ISSUE TITLE
Backend Route Typing

### ISSUE THREAD

**Issue by Gregoor** [mozilla/voice-web#835]
Depends on #834 

I recently stumbled over [restyped](https://github.com/rawrmaan/restyped) which might be a great addition here, as it would allow us to type across the backend-frontend "line".

Unfortunately this doesn't do runtime type checks for route parameter. And while technically possible, it doesn't generate docs based on the type information. It'd be great to have a library that does all of that. Think GraphQL for REST, which I guess would be Swagger?
Happy to get some input on this!

**Comment by Gregoor** [comment]
Unfortunately it looks like restyped [wouldn't allow us to nest routers](https://github.com/rawrmaan/restyped-express-async/issues/3), so I think we'll stall on getting stricter typing in.

We also talked about looking into GraphQL, which imo solves a lot of our current problems (namely docs, typing and data fetching). For now we postponed moving on it, but we might look into it again when working on views that have more diverse data requirements.
